src/generator/named-types.ts

```diff
@@ -194,5 +225,22 @@ function recursiveVisitMembersType(
 
   if (isOptionalType(valueType)) {
     recursiveVisitMembersType(valueType.wrappedType, visit, `${path}`);
+    return;
   }
+
+  if (isUnionType(valueType)) {
+    const subType = basicTypeOfUnion(valueType);
```

Comment:
nit: move the conversion of union type to enum type to `visit` for aligning the behaviour with how the tuple type is converted to interface type.

Reply:
You mean this?
<img width="499" alt="image" src="https://github.com/microsoft/ts-gyb/assets/24806909/7164c04e-6906-4e7e-928a-3a23ca05cfad">

Reply:
I meant moving this chunk of code here: https://github.com/microsoft/ts-gyb/blob/55f9d4103e0caf11494c0189049f8bb590d11e5f/src/generator/named-types.ts#L117C1-L123C12 so the two conversions are placed together.

It's important to note that the name of this function is `recursivelyVisitMembersType` which implies it doesn't modify anything. That's why the conversation from tuple type to interface type is placed inside the closure `visit`.